gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/publisher/HiveConvertPublisher.java

```diff
@@ -83,6 +91,10 @@ public HiveConvertPublisher(State state) throws IOException {
     } catch (SQLException e) {
       throw new RuntimeException(e);
     }
+
+    this.watermarker =
```

Comment:
I think we should encode the fact that you pass a state to instantiate a the watermark class in an interface

interface HiveSourceWatermarkerFactory {
   HiveSourceWatermarker createFromState(State state);
}

What we configure is the factory class which gets created using the default constructor.

I think the above approach documents more explicitly the interface and we can also add different factory methods in the future.